### PR TITLE
Fix ui tests

### PR DESCRIPTION
**Pull request by catehstn**
Fixed the getHTMLContent function - now scrolls the toolbar to tap the appropriate button.
Is whether or not the toolbar is expanded by default a device setting? Would probably enable that by default for the demo as it makes this easier.

Added `<p>` and `</p>` tags to get tests to pass. If this behaviour changes, these tests will need to be updated. Comments out two tests that need more change (and are maybe surfacing bugs).

Uncommented out the list tests, minor changes to get those working.

Stripping out all whitespace characters between tags as we should not IMO be testing that. Open to feedback here.

To test: Open example app, and run the tests. Confirm that they pass.

### CODE REVIEW

**Comment by rachelmcr** [comment]
>Fixed the getHTMLContent function - now scrolls the toolbar to tap the appropriate button.

I just reread this note; were you having trouble with this function not working as expected before? XCUITest should always scroll in scrollViews to the element you're trying to tap, without having to explicitly swipe.

File: Example/AztecUITests/AztecUITests.swift [hunk 1]
```diff
@@ -138,10 +134,12 @@ class AztecSimpleTextFormattingTests: XCTestCase {
         enterTextInField(text: "\n2")
 
         let text = getHTMLContent()
-        let expected = "1<br><hr><br>2"
+        let expected = "<p>1</p><p><hr></p><p>2</p>"
         XCTAssertEqual(expected, text)
     }
 
+    /*
+     Commenting these out because they fails: Why is the more tag wrapped in a blockquote?
```
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply Example/AztecUITests/AztecUITests.swift]: Some extra info on this: the wrapping we're doing here (I mean the output we provide given the input) is not great.  I was aware of this but failed to track this issue.

I think the outputs should be:

1: `<!--more-->`
2: `<p>1</p><!--more><p>2</p>`

I've now opened an issue to track this problem, so I'd consider leaving these disabled (and we'll turn them on when that issue is resolved).

File: Example/AztecUITests/XCTest+Extensions.swift [hunk 2]
```diff
@@ -61,10 +65,21 @@ extension XCTest {
      */
     func getHTMLContent() -> String {
         let app = XCUIApplication()
-
-        app.buttons[elementStringIDs.sourcecodeButton].tap()
-        let htmlContentTextView = app.textViews[elementStringIDs.htmlTextField]
+        
+        // Expects the format bar to be expanded.
+        let elementsQuery = app.scrollViews.otherElements
+        elementsQuery.buttons[elementStringIDs.mediaButton].swipeLeft()
```
[the remark this answers is not in the document]
Comment (brbrr) [reply Example/AztecUITests/XCTest+Extensions.swift]: > the test will swipe as needed to get to the toolbar button it's trying to tap on
That is strange, but it isn't working like that on iPhone SE but works fine for iPhone 8

**Comment by catehstn** [comment]
@rachelmcr yes - this wasn't working for me.

Also, we need to handle the case where the toolbar isn't expanded. The button is missing an a11y label. IMO though, we should just always expand the toolbar in the example app.

For now, I updated the comment for accuracy, and added a check which only scrolls if the button isn't there. This fixed the blockquote thing, but because the behaviour will change in #818 I left it as-is for now.

**Comment by frosty** [comment]
> Is whether or not the toolbar is expanded by default a device setting? Would probably enable that by default for the demo as it makes this easier.

Yes, we store the expanded status in `NSUserDefaults`, and the toolbar checks on creation to see if it should display expanded or not – so it'd be easy enough for it to always be expanded. I guess even if we defaulted the demo app to always expanded, we'd ideally want tests to verify that it expands / contracts correctly.

**Comment by catehstn** [comment]
@frosty - will need to figure out a11y for the states then "more toolbar options" / "fewer toolbar options"? But regardless, IMO better to have one test that checks compressing / expanding rather than needing to expand it every test.

**Comment by catehstn** [comment]
@diegoreymendez can you merge this please? <3

**Comment by diegoreymendez** [comment]
Awesome contribution @catehstn.  Thank you! 👍

**Pull request merged**